auth.go

```diff
@@ -387,36 +404,49 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 // Create a new session. Returns a session key, which can be used in future to retrieve the token.
 // The internal session expiry is controlled with the member NewSessionExpiresAfter.
 // The session key is typically sent to the client as a cookie.
-func (x *Central) Login(identity, password string) (sessionkey string, token *Token, e error) {
-	token = &Token{}
-	if token.Identity, e = x.authenticator.Authenticate(identity, password); e == nil {
-		x.Log.Infof("Login authentication success (%v)", identity)
-		var permit *Permit
-		if permit, e = x.permitDB.GetPermit(identity); e == nil {
-			if x.MaxActiveSessions != 0 {
-				if e = x.sessionDB.InvalidateSessionsForIdentity(identity); e != nil {
-					x.Log.Errorf("Invalidate sessions for identity (%v) failed when enforcing MaxActiveSessions (%v)", identity, e)
-					return "", nil, e
-				}
-			}
-			token.Expires = time.Now().Add(x.NewSessionExpiresAfter)
-			token.Permit = *permit
-			sessionkey = generateSessionKey()
-			if e = x.sessionDB.Write(sessionkey, token); e == nil {
-				x.Stats.IncrementGoodLogin(x.Log)
-				x.Log.Infof("Login successful (%v)", identity)
-				return
-			}
-		} else {
-			x.Log.Infof("Login GetPermit failed (%v) (%v)", identity, e)
-		}
-	} else {
+func (x *Central) Login(identity, password string) (string, *Token, error) {
```

Comment:
In a function definition such as this, you _must_ name the return values, because it's not at all obvious to a reader of this function definition, what the first "string" return value holds. It's an eternal rule of any function that if you need to read it's body to determine what it does, then you've failed to create a good API. As a litmus test, you can keep "godoc" running, and make sure that you can effectively read an API just by looking at the godoc page that is generated for that API.